index.js

```diff
@@ -10,27 +10,52 @@
  */
 let store = {};
 
-const _callsubscribers = (key, newValue, prevValue) => {
-  store[key].subscribers.forEach(callback => {
-    if (typeof callback === 'function') {
-      callback(newValue, prevValue);
+// Thanks to https://stackoverflow.com/a/25456134/1954789
+export function deepEqual(x, y) {
+  // TODO: this will not handle cyclical references !
+  // TODO: this does not check class name !
+
+  if (x === y) {
+    return true;
+  }
+
+  if ((typeof x == "object" && x != null) && (typeof y == "object" && y != null)) {
+    if (Object.keys(x).length != Object.keys(y).length) {
+      return false;
     }
-  });
-};
+
+    for (const prop in x) {
+      if (!y.hasOwnProperty(prop)) {
+        return false;
+      }
+      if (!deepEqual(x[prop], y[prop])) {
+        return false;
+      }
+    }
+
+    return true;
+  }
+
+  return false;
+}
 
 export default {
-  set (key, value) {
-    let currentValue = undefined;
+  set(key, newValue) {
+    let prevValue = undefined;
 
     if (!store[key]) {
-      store[key] = { value, subscribers: [] };
+      store[key] = { value: newValue, subscribers: [] };
     } else {
-      currentValue = store[key].value;
-      store[key].value = value;
+      prevValue = store[key].value;
+      if (deepEqual(newValue, prevValue)) {
+        // If we have a previous value, and if it is the same,
+        // then we don't notify
+        return;
+      }
+      store[key].value = newValue;
     }
 
-    // TODO: Call callback only if the value really changed
-    _callsubscribers(key, value, currentValue);
+    store[key].subscribers.forEach(callback => callback(newValue, prevValue));
```

Comment:
We should move this line a few lines above, at the end of the prev `else`. Because it doesn't make sense to try to call the subscribers if `!store[key]` (due to there are not subscribers)

Reply:
delivered